File: src/main/java/net/insprill/cjm/hooks/SuperVanishHook.java
package net.insprill.cjm.hooks;

import de.myzelyam.api.vanish.PlayerVanishStateChangeEvent;
import net.insprill.cjm.handlers.VanishHandler;
import org.bukkit.Bukkit;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.EventPriority;
import org.bukkit.event.Listener;

public class SuperVanishHook implements Listener {

    @EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
    public void onVanishToggle(PlayerVanishStateChangeEvent e) {
        Player player = Bukkit.getPlayer(e.getUUID());
        if (player == null)
            return; // Why the hell can it return a UUID for a player that doesn't exist??
        VanishHandler.handleVanishToggle(player, e.isVanishing());
    }

}

File: src/main/kotlin/net/insprill/cjm/compatibility/hook/SuperVanishHook.kt
package net.insprill.cjm.compatibility.hook

import de.myzelyam.api.vanish.PlayerVanishStateChangeEvent
import net.insprill.cjm.CustomJoinMessages
import org.bukkit.Bukkit
import org.bukkit.event.EventHandler
import org.bukkit.event.EventPriority
import org.bukkit.event.Listener

class SuperVanishHook(private val plugin: CustomJoinMessages) : Listener {

    @EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
    fun onVanishToggle(e: PlayerVanishStateChangeEvent) {
        val player = Bukkit.getPlayer(e.uuid) ?: return // Why the hell can it return a UUID for a player that doesn't exist??
        plugin.vanish.handleToggle(player, e.isVanishing)
    }

}